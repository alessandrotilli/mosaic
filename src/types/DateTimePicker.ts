import { ILocalizable } from "./Base";

export type DateTimePickerType = ILocalizable & {
  label: string;
  value?: Date | string | number;
  onAccept?: (value: Date | undefined | null) => void;
  ampm?: boolean;
  mobileView?: boolean;
  timeZone?: string;
  format?: string;
  timeView?: "hrs" | "min" | "sec";
};

export type viewType = "year" | "month" | "day" | "hours" | "minutes" | "seconds";
