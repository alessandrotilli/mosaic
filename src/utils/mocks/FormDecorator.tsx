import { useCallback, useEffect, useState } from "react";
import { DecoratorFn } from "@storybook/react";

export type FormValue = boolean | number | string | null;

const FormDecorator: DecoratorFn = (Story, { args: { onChange: externalOnChange, value: externalValue, ...args } }) => {
  const [value, setValue] = useState(externalValue);

  useEffect(() => {
    if (externalValue !== value) {
      setValue(externalValue);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [externalValue]);

  const onChange = useCallback(
    (value: FormValue) => {
      externalOnChange && externalOnChange(value);
      setValue(value);
    },
    [externalOnChange]
  );

  return Story({
    args: {
      ...args,
      onChange,
      value,
    },
  });
};

export default FormDecorator;
