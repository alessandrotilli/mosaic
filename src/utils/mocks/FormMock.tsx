import React, { cloneElement, FC, PropsWithChildren, ReactElement, useMemo, useState } from "react";
import { DecoratorFn } from "@storybook/react";

type IFormValue = boolean | number | string | null;

interface FormMockType {
  onChangePropName?: string;
  value: IFormValue;
  valuePropName?: string;
}

const FormMock: FC<PropsWithChildren<FormMockType>> = ({
  children,
  onChangePropName = "onChange",
  value: externalValue,
  valuePropName = "value",
}) => {
  const [value, setValue] = useState(externalValue);

  const wrappedFormElement = useMemo(
    () =>
      cloneElement(children as ReactElement<any>, {
        [onChangePropName]: setValue,
        [valuePropName]: value,
      }),
    [children, onChangePropName, value, valuePropName]
  );

  return wrappedFormElement;
};

const formDecorator: DecoratorFn = (Story, { args }) => <FormMock value={args.value}>{Story()}</FormMock>;

export { formDecorator, FormMock };
