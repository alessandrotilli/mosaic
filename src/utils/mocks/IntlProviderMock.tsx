import React, { FC } from "react";
import { IntlProvider } from "react-intl";

export enum LocaleMock {
  en = "en",
  it = "it-IT",
}

export enum MessageMock {
  button = "button",
  cancel = "cancel",
  checkbox = "checkbox",
  confirm = "confirm",
  inputNumber = "inputNumber",
  inputText = "inputText",
  subtitle = "subtitle",
  switch = "switch",
  title = "title",
  typography = "typography",
}

interface IntlProviderMockType {
  locale?: LocaleMock;
}

export const mockedMessages = {
  [LocaleMock.en]: {
    [MessageMock.button]: "Button",
    [MessageMock.cancel]: "Cancel",
    [MessageMock.checkbox]: "Checkbox",
    [MessageMock.confirm]: "Confirm",
    [MessageMock.inputNumber]: "Input Number",
    [MessageMock.inputText]: "Input Text",
    [MessageMock.subtitle]: "Subtitle",
    [MessageMock.switch]: "Switch",
    [MessageMock.title]: "Title",
    [MessageMock.typography]: "Typography",
  },
  [LocaleMock.it]: {
    [MessageMock.button]: "Bottone",
    [MessageMock.cancel]: "Annulla",
    [MessageMock.checkbox]: "Spunta",
    [MessageMock.confirm]: "Conferma",
    [MessageMock.inputNumber]: "Numero in input",
    [MessageMock.inputText]: "Testo in input",
    [MessageMock.subtitle]: "Sottotitolo",
    [MessageMock.switch]: "Interruttore",
    [MessageMock.title]: "Titolo",
    [MessageMock.typography]: "Tipografia",
  },
};

const IntlProviderMock: FC<IntlProviderMockType> = ({ children, locale = LocaleMock.en }) => {
  return (
    <IntlProvider locale={locale} messages={mockedMessages[locale]}>
      {children}
    </IntlProvider>
  );
};

export default IntlProviderMock;
