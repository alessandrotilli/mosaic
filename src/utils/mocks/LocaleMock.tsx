import React from "react";
import { DecoratorFn } from "@storybook/react";

import { LocalizedContextProvider } from "../../contexts/Localized";

export type ILocaleMock = "en" | "it-IT";

export enum MessageMock {
  button = "intl.message.button",
  cancel = "intl.message.cancel",
  checkbox = "intl.message.checkbox",
  confirm = "intl.message.confirm",
  inputNumber = "intl.message.inputNumber",
  inputText = "intl.message.inputText",
  subtitle = "intl.message.subtitle",
  switch = "intl.message.switch",
  title = "intl.message.title",
  typography = "typography",
}

type IMessagesMock = {
  [key in ILocaleMock]: {
    [key in MessageMock]: string;
  };
};

export const MESSAGES_MOCK: IMessagesMock = {
  en: {
    [MessageMock.button]: "Button",
    [MessageMock.cancel]: "Cancel",
    [MessageMock.checkbox]: "Checkbox",
    [MessageMock.confirm]: "Confirm",
    [MessageMock.inputNumber]: "Input Number",
    [MessageMock.inputText]: "Input Text",
    [MessageMock.subtitle]: "Subtitle",
    [MessageMock.switch]: "Switch",
    [MessageMock.title]: "Title",
    [MessageMock.typography]: "Typography",
  },
  "it-IT": {
    [MessageMock.button]: "Bottone",
    [MessageMock.cancel]: "Annulla",
    [MessageMock.checkbox]: "Spunta",
    [MessageMock.confirm]: "Conferma",
    [MessageMock.inputNumber]: "Numero in input",
    [MessageMock.inputText]: "Testo in input",
    [MessageMock.subtitle]: "Sottotitolo",
    [MessageMock.switch]: "Interruttore",
    [MessageMock.title]: "Titolo",
    [MessageMock.typography]: "Tipografia",
  },
};

const getLocalizedMessage = (key: MessageMock, locale: ILocaleMock): string => MESSAGES_MOCK[locale][key];

const localeDecorator: DecoratorFn = (Story) => (
  <LocalizedContextProvider localize={(key) => getLocalizedMessage(key as MessageMock, "it-IT")}>
    {Story()}
  </LocalizedContextProvider>
);

export { getLocalizedMessage, localeDecorator };
