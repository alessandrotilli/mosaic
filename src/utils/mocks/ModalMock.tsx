/* eslint-disable @typescript-eslint/no-explicit-any */
import React, {
  cloneElement,
  FC,
  PropsWithChildren,
  ReactElement,
  useCallback,
  useEffect,
  useMemo,
  useState,
} from "react";
import { DecoratorFn } from "@storybook/react";

import Button from "../../components/Button";
import { Icons } from "../../types/Icon";

interface ModalMockType {
  buttonLabel?: string;
  open?: boolean;
}

const ModalMock: FC<PropsWithChildren<ModalMockType>> = ({
  buttonLabel = "Open",
  children,
  open: externalOpen = false,
}) => {
  const [open, setOpen] = useState(externalOpen);

  useEffect(() => {
    if (externalOpen !== open) {
      setOpen(externalOpen);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [externalOpen]);

  const toggleModal = useCallback(() => setOpen((prevOpen) => !prevOpen), [setOpen]);

  const wrappedModal = useMemo(
    () =>
      cloneElement(children as ReactElement<any>, {
        closable: true,
        onClose: toggleModal,
        open,
      }),
    [children, open, toggleModal]
  );

  return (
    <div>
      <Button icon={{ name: Icons.open_new }} label={buttonLabel} onClick={toggleModal} variant="outlined" />
      {wrappedModal}
    </div>
  );
};

const modalDecorator: DecoratorFn = (Story) => <ModalMock>{Story()}</ModalMock>;

export { modalDecorator, ModalMock };
