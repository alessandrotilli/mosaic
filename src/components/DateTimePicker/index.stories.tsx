import { Meta, StoryObj } from "@storybook/react";

import { localeDecorator } from "../../utils/mocks/LocaleMock";
import getDocsPage from "../../utils/stories";

import DateTimePicker, { DATA_CY_DEFAULT, DATA_CY_SHORTCUT, LOCALIZABLE_PROPS, LocalizedDateTimePicker } from ".";

const COMPONENT_NAME = "DateTimePicker";
DateTimePicker.displayName = COMPONENT_NAME;

const meta = {
  title: "Inputs/DateTimePicker",
  component: LocalizedDateTimePicker,
  decorators: [localeDecorator],
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI DateTimePicker Component",
          url: "https://mui.com/x/react-date-pickers/date-time-picker/",
        },
        component: COMPONENT_NAME,
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
          dataCyShortcut: DATA_CY_SHORTCUT,
        },
        localizableProps: LOCALIZABLE_PROPS,
      }),
    },
  },
} satisfies Meta<typeof LocalizedDateTimePicker>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    label: "Date Time",
  },
};

export const HoursView: Story = {
  args: {
    ...Primary.args,
    timeView: "hrs",
  },
};

export const SecondsView: Story = {
  args: {
    ...Primary.args,
    timeView: "sec",
  },
};

export const MobileView: Story = {
  args: {
    ...SecondsView.args,
    mobileView: true,
    timeZone: "utc+8",
    value: "2023-10-29T00:00:00.000+02:00",
  },
};

export const FormatAmPm: Story = {
  args: {
    ...Primary.args,
    ampm: true,
  },
};

export const Localized: Story = {
  args: {
    ...Primary.args,
    localized: true,
    label: "locale.dateTime",
  },
};

export const Date = {
  args: {
    ...Primary.args,
    type: "date",
  },
};

export const Time: Story = {
  args: {
    ...Primary.args,
    type: "time",
  },
};
