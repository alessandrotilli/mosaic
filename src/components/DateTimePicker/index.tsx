import React, { FC, useCallback, useMemo } from "react";
import { AdapterLuxon } from "@mui/x-date-pickers/AdapterLuxon";
import { DateTimePicker as MuiDateTimePicker } from "@mui/x-date-pickers/DateTimePicker";
import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider";
import { DateTime } from "luxon";

import { useMosaicContext } from "../../hooks/useMosaicContext";
import { DateTimePickerType, viewType } from "../../types/DateTimePicker";
import { logError } from "../../utils/logger";

export const DATA_CY_DEFAULT = "timePicker";

const DateTimePicker: FC<DateTimePickerType> = ({
  dataCy = DATA_CY_DEFAULT,
  label = "Date Time",
  value,
  onAccept,
  ampm = false,
  mobileView,
  timeView = "min",
  format,
  timeZone,
}: DateTimePickerType) => {
  const { timeZone: contextTimeZone } = useMosaicContext();

  const desctopMode = useMemo(() => {
    return mobileView ? "Mobile" : undefined;
  }, [mobileView]);

  const views: viewType[] = useMemo(() => {
    if (timeView === "hrs") {
      return ["year", "month", "day", "hours"];
    }
    if (timeView === "sec") {
      return ["year", "month", "day", "hours", "minutes", "seconds"];
    }
    return ["year", "month", "day", "hours", "minutes"];
  }, [timeView]);

  const zone = useMemo(() => {
    return timeZone ? timeZone : contextTimeZone;
  }, [timeZone, contextTimeZone]);

  const isDate = useCallback((date: Date) => {
    const valueDateTime = DateTime.fromJSDate(date);
    return valueDateTime.isValid;
  }, []);

  const dateTimeValue = useMemo(() => {
    if (value) {
      const valueDate = new Date(value);
      const validDate = isDate(valueDate);
      if (validDate) {
        return DateTime.fromJSDate(valueDate);
      }
      logError("DateTimePicker", "Invalid Date");
      return undefined;
    }
  }, [value, isDate]);

  const onAcceptIso = useCallback(
    (value: DateTime | undefined | null) => {
      if (onAccept) {
        onAccept(value?.toISO());
      }
    },
    [onAccept]
  );

  return (
    <LocalizationProvider dateAdapter={AdapterLuxon}>
      <MuiDateTimePicker
        timezone={zone}
        desktopModeMediaQuery={desctopMode}
        value={dateTimeValue}
        onAccept={onAcceptIso}
        data-cy={dataCy}
        label={label}
        views={views}
        format={format}
        ampm={ampm}
      />
    </LocalizationProvider>
  );
};

export default DateTimePicker;
