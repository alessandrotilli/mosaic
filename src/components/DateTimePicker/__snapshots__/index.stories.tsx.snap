// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs/DateTimePicker Date smoke-test 1`] = `
<div class="MuiFormControl-root MuiTextField-root css-z3c6am-MuiFormControl-root-MuiTextField-root">
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="false"
         for=":rd:"
         id=":rd:-label"
  >
    Date Time
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           autocomplete="off"
           id=":rd:"
           placeholder="MM/DD/YYYY"
           type="text"
           inputmode="text"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
           value
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root">
      <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
              aria-label="Choose date"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="CalendarIcon"
        >
          <path d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z">
          </path>
        </svg>
      </button>
    </div>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-yjsfm1">
        <span>
          Date Time
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/DateTimePicker FormatAmPm smoke-test 1`] = `
<div class="MuiFormControl-root MuiTextField-root css-z3c6am-MuiFormControl-root-MuiTextField-root">
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="false"
         for=":r9:"
         id=":r9:-label"
  >
    Date Time
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           autocomplete="off"
           id=":r9:"
           placeholder="MM/DD/YYYY hh:mm aa"
           type="text"
           inputmode="text"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
           value
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root">
      <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
              aria-label="Choose date"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="CalendarIcon"
        >
          <path d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z">
          </path>
        </svg>
      </button>
    </div>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-yjsfm1">
        <span>
          Date Time
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/DateTimePicker HoursView smoke-test 1`] = `
<div class="MuiFormControl-root MuiTextField-root css-z3c6am-MuiFormControl-root-MuiTextField-root">
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="false"
         for=":r2:"
         id=":r2:-label"
  >
    Date Time
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           autocomplete="off"
           id=":r2:"
           placeholder="MM/DD/YYYY hh"
           type="text"
           inputmode="text"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
           value
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root">
      <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
              aria-label="Choose date"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="CalendarIcon"
        >
          <path d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z">
          </path>
        </svg>
      </button>
    </div>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-yjsfm1">
        <span>
          Date Time
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/DateTimePicker Localized smoke-test 1`] = `
<div class="MuiFormControl-root MuiTextField-root css-z3c6am-MuiFormControl-root-MuiTextField-root">
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="false"
         for=":rb:"
         id=":rb:-label"
  >
    Data Ore
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           autocomplete="off"
           id=":rb:"
           placeholder="MM/DD/YYYY hh:mm"
           type="text"
           inputmode="text"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
           value
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root">
      <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
              aria-label="Choose date"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="CalendarIcon"
        >
          <path d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z">
          </path>
        </svg>
      </button>
    </div>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-yjsfm1">
        <span>
          Data Ore
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/DateTimePicker MobileView smoke-test 1`] = `
<div class="MuiFormControl-root MuiTextField-root css-z3c6am-MuiFormControl-root-MuiTextField-root">
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="true"
         for=":r7:"
         id=":r7:-label"
  >
    Date Time
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl Mui-readOnly MuiInputBase-readOnly css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           autocomplete="off"
           id=":r7:"
           placeholder="MM/DD/YYYY hh:mm:ss"
           readonly
           type="text"
           aria-label="Choose date, selected date is Oct 29, 2023"
           inputmode="text"
           class="MuiInputBase-input MuiOutlinedInput-input Mui-readOnly MuiInputBase-readOnly css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
           value="10/29/2023 06:00:00"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-14lo706">
        <span>
          Date Time
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/DateTimePicker Primary smoke-test 1`] = `
<div class="MuiFormControl-root MuiTextField-root css-z3c6am-MuiFormControl-root-MuiTextField-root">
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="false"
         for=":r0:"
         id=":r0:-label"
  >
    Date Time
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           autocomplete="off"
           id=":r0:"
           placeholder="MM/DD/YYYY hh:mm"
           type="text"
           inputmode="text"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
           value
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root">
      <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
              aria-label="Choose date"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="CalendarIcon"
        >
          <path d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z">
          </path>
        </svg>
      </button>
    </div>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-yjsfm1">
        <span>
          Date Time
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/DateTimePicker SecondsView smoke-test 1`] = `
<div class="MuiFormControl-root MuiTextField-root css-z3c6am-MuiFormControl-root-MuiTextField-root">
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="false"
         for=":r4:"
         id=":r4:-label"
  >
    Date Time
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           autocomplete="off"
           id=":r4:"
           placeholder="MM/DD/YYYY hh:mm:ss"
           type="text"
           inputmode="text"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
           value
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root">
      <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
              aria-label="Choose date"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="CalendarIcon"
        >
          <path d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z">
          </path>
        </svg>
      </button>
    </div>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-yjsfm1">
        <span>
          Date Time
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/DateTimePicker Time smoke-test 1`] = `
<div class="MuiFormControl-root MuiTextField-root css-z3c6am-MuiFormControl-root-MuiTextField-root">
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
         data-shrink="false"
         for=":rf:"
         id=":rf:-label"
  >
    Date Time
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root">
    <input aria-invalid="false"
           autocomplete="off"
           id=":rf:"
           placeholder="hh:mm"
           type="text"
           inputmode="text"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
           value
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root">
      <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
              aria-label="Choose time"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="ClockIcon"
        >
          <path d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z">
          </path>
          <path d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z">
          </path>
        </svg>
      </button>
    </div>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend class="css-yjsfm1">
        <span>
          Date Time
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
